src/aggregate/aggregate_request.c

```diff
@@ -452,6 +443,7 @@ int PLNGroupStep_AddReducer(PLN_GroupStep *gstp, const char *name, ArgsCursor *a
   gr->name = name;
   int rv = AC_GetVarArgs(ac, &gr->args);
   if (rv != AC_OK) {
+    array_pop(gstp->reducers);
```

Comment:
should check if NULL before freeing instead...

Reply:
You will have a half baked element, its just better to handle it here where we know exactly what was allocated